src/Zcu/PerThread.zig

```diff
@@ -1078,6 +1081,11 @@ fn semaDecl(pt: Zcu.PerThread, decl_index: Zcu.Decl.Index) !Zcu.SemaDeclResult {
     defer tracy.end();
 
     const zcu = pt.zcu;
+    const comp = zcu.comp;
+
+    comp.time_trace.event(.{ .decl = decl_index }, pt);
+    defer comp.time_trace.event(.end, pt);
```

Comment:
let's not have both tracy and time trace in the same function. max 1 tracing per function. if a function needs both then make an abstraction that lowers to one or the other depending on build configuration